### ISSUE TITLE
Create spack-stack unified-dev install on Discover from develop (aka `1.5.1+`)

### ISSUE THREAD

**Issue by climbfuji** [JCSDA/spack-stack#878]
**Is your feature request related to a problem? Please describe.**
The NASA team on Discover needs the new `gmao-swell-env` in the unified-env. This environment doesn't exist in `spack-stack@1.5.1` yet, but in develop. This is also needed so that JEDI (ALGO) can proceed with merging updates that require `ecmwf-atlas@0.35.0:`.

**Describe the solution you'd like**
Since release 1.5.1 was merged back last week, we can install from develop to get a slightly newer setup than 1.5.1. This environment will provide everything needed for NASA and for the JEDI teams to proceed.

**Additional context**
n/a

**Comment by climbfuji** [comment]
@danholdaway @shlyaeva I created the new environment on Discover, see spack-stack PR https://github.com/JCSDA/spack-stack/pull/879 for the documentation update. In short:
```
The following is required for building new spack environments and for using spack to build and run software.

.. code-block:: console

   module purge
   module use /discover/swdev/jcsda/spack-stack/modulefiles
   module load miniconda/3.9.7
   module load ecflow/5.8.4
   module load mysql/8.0.31

For ``spack-stack-1.5.1`` with Intel, load the following modules after loading miniconda and ecflow:

.. code-block:: console

   module use /gpfsm/dswdev/jcsda/spack-stack/spack-stack-dev-20231114/envs/unified-env/install/modulefiles/Core
   module load stack-intel/2022.0.1
   module load stack-intel-oneapi-mpi/2021.5.0
   module load stack-python/3.10.8
   module available

For ``spack-stack-1.5.1`` with GNU, load the following modules after loading miniconda and ecflow:

.. code-block:: console

   module use /gpfsm/dswdev/jcsda/spack-stack/spack-stack-dev-20231114/envs/unified-env/install/modulefiles/Core
   module load stack-gcc/10.1.0
   module load stack-openmpi/4.1.3
   module load stack-python/3.10.8
   module available
```
Can you please give this software stack a try and let me know how it goes?

**Comment by shlyaeva** [comment]
Thank you @climbfuji!
I don't have an account on discover, so can't do it unfortunately. @danholdaway would you or someone else at GMAO have time to test? We'd like to start imposing the higher versions of eckit/atlas that are available in 1.5.1.

**Comment by shlyaeva** [comment]
Note: we plan to start merging PRs requiring higher versions of eckit and atlas available in this version of stack next Thursday Nov 30th (https://github.com/orgs/JCSDA-internal/discussions/74).
